purchase_all_shipments/model/purchase_order.py

```diff
@@ -0,0 +1,61 @@
+#    Author: Leonardo Pistone
+#    Copyright 2015 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from openerp import models, fields, api
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    def _all_shipment_count(self):
+        self.all_shipment_count = len(self.all_picking_ids)
+
+    def _all_pickings(self):
+        groups = self.env['procurement.group']
+        for pick in self.picking_ids:
+            for move in pick.move_lines:
+                groups |= move.group_id
+
+        all_moves = self.env['stock.move'].search(
+            [('group_id', 'in', groups.mapped('id'))]
+        )
+        self.all_picking_ids = all_moves.mapped('picking_id')
+
+    @api.multi
+    def view_all_picking(self):
+        """Similar to the view_picking method in the purchase module"""
+        action_data = self.env.ref('stock.action_picking_tree').read()[0]
```

Comment:
It seems correct to me, the `[0]` is necessary as `read()` on record set returns a list.